core/modules/query/QueryContext.cc

```
@@ -52,35 +155,65 @@ QueryContext::resolve(std::shared_ptr<ColumnRef> cr) {
             if (concrete.db.empty()) {
                 concrete.db = defaultDb;
             }
-            return concrete;
+            dbTableSet.insert(concrete);
+            return dbTableSet;
         }
     }
     // Set default db and table.
     DbTablePair p;
-    if (cr->table.empty()) { // No db or table: choose first resolver pair
-        p = resolverTables[0];
-        // TODO: We can be fancy and check the column name against the
-        // schema for the entries on the resolverTables, and choose
-        // the matching entry.
+    if (cr->table.empty()) { // No db or table.
+        if (columnToTablesMap.empty()) {
+            // This should only be the case if no connection to the database could be made, which
+            // should only happen in unit tests. This should get through the unit tests and have
```

I'm not proposing to eliminate unit tests, I'm just saying that comment is not entirely correct, there are cases when connection to database may fail and `columnToTablesMap` will be left empty. I feel that we don't handle this case quite correctly yet, right now it's a silent failure which I do not quite like.